syntax = "proto3";

package envoy.config.resource_monitor.unused_heap.v2alpha;

import "envoy/type/percent.proto";
import "validate/validate.proto";

option java_package = "io.envoyproxy.envoy.config.resource_monitor.unused_heap.v2alpha";
option java_outer_classname = "UnusedHeapProto";
option java_multiple_files = true;

// [#protodoc-title: Unused heap]
// [#extension: envoy.resource_monitors.unused_heap]

// The unused heap resource monitor reports the amount of memory reserved by TCMalloc, but not used
// by Envoy, computed as a fraction of currently unused heap memory divided by a statically
// configured maximum specified in the UnusedHeapConfig.
message UnusedHeapConfig {
  oneof monitor_type {
    uint64 max_unused_heap_size_bytes = 1 [(validate.rules).uint64 = {gt: 0}];
    type.Percent max_unused_heap_percent = 2;
  }
}
